### PR TITLE
Refactor/simplify window frame utils

### PR DESCRIPTION
**Pull request by ozankabak**
## Which issue does this PR close?

Closes #.

## Rationale for this change

Reducing API clutter for checking/verifying/regularizing window frames.

## What changes are included in this PR?

Apart from the said API simplification, fixes the window fuzz tests to make it clear when streaming execution is not possible. Also removes some unwrap calls from window related code.

## Are these changes tested?

Yes

## Are there any user-facing changes?

A very minor change at the API level: The `regularize` function becomes a method of `WindowFrame`

### CODE REVIEW

File: datafusion/proto/src/logical_plan/from_proto.rs [hunk 1]
```diff
@@ -353,15 +347,15 @@ pub fn parse_expr(
...
-                    Ok(Expr::WindowFunction(WindowFunction::new(
+                    Expr::WindowFunction(WindowFunction::new(
                         expr::WindowFunctionDefinition::AggregateUDF(udaf_function),
                         args,
                     ))
                     .partition_by(partition_by)
                     .order_by(order_by)
                     .window_frame(window_frame)
                     .build()
-                    .unwrap())
```
Comment (alamb) [inline datafusion/proto/src/logical_plan/from_proto.rs:364]: this is a nice improvement too

File: datafusion/expr/src/window_frame.rs [hunk 2]
```diff
@@ -246,59 +246,55 @@ impl WindowFrame {
...
-pub fn check_window_frame(frame: &WindowFrame, order_bys: usize) -> Result<()> {
-    if frame.units == WindowFrameUnits::Range && order_bys != 1 {
-        // See `regularize_window_order_by`.
-        if !(frame.start_bound.is_unbounded()
-            || frame.start_bound == WindowFrameBound::CurrentRow)
-            || !(frame.end_bound == WindowFrameBound::CurrentRow
-                || frame.end_bound.is_unbounded())
-        {
-            plan_err!("RANGE requires exactly one ORDER BY column")?
+    /// Returns whether the window frame can accept multiple ORDER BY expressons.
```
Comment (alamb) [inline datafusion/expr/src/window_frame.rs:281]: I agree this is much nicer

File: datafusion/expr/src/window_frame.rs [hunk 3]
```diff
@@ -246,59 +246,55 @@ impl WindowFrame {
...
+                // If an ORDER BY clause is absent, it is equivalent to an
+                // ORDER BY clause with constant value as sort key. If an
+                // ORDER BY clause is present but has more than one column,
+                // it is unchanged. Note that this follows PostgreSQL behavior.
+                if order_by.is_empty() {
+                    order_by.push(Expr::Sort(Sort::new(
+                        Box::new(Expr::Literal(ScalarValue::UInt64(Some(1)))),
+                        true,
+                        false,
+                    )));
```
Comment (alamb) [inline datafusion/expr/src/window_frame.rs:267]: FWIW you can write this more succinctly with the expr API like this I think

```suggestion
                    order_by.push(lit(1u64).sort(true, false));
```

**Review by alamb — APPROVED** [review]
Thank you @ozankabak  -- this looks like a nice improvement to me

File: datafusion/expr/src/window_frame.rs (hunk 3 above)
Comment (ozankabak) [reply datafusion/expr/src/window_frame.rs]: Indeed much nicer, thank you -- added

**Review by mustafasrepo — APPROVED** [review]
LGTM!.

**Pull request merged**